### PR TITLE
feat: add-bfs-check

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Added a BFS-like check for FSM specifications. This is required because the previous checks did not catch all possible erroneous specifications. Namely, a specification like

start_states: [A]

(A, ev1): B
(B, ev1): C
(D, ev1): A
(X, ev1): Y
(Y, ev1): X

would have passed the checks, but it clearly contains 2 disjoint components {A,B,C} and {X,Y}, and the latter one is unreachable, yet it does not have any orphan state. Thus, I removed the concept of "orphan state", as it is grouped into the more general concept of "reachable state".

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/analyse/abci/app_spec.py [hunk 1]
```diff
@@ -205,6 +207,32 @@ def generate(self) -> Dict[str, Any]:
...
+        graph: Dict[str, Set[str]] = defaultdict(set)
+        for (s1, _), s2 in transition_func.items():
+            graph[s1].add(s2)
+
+        visited: Set[str] = start_states.copy()
+        queue: List[str] = []
+        queue.extend(list(start_states))
+
+        while queue:
+            s = queue.pop(0)
```
Comment (Karrenbelt) [inline autonomy/analyse/abci/app_spec.py:227]: nice nice. Popping first element of a list is no good in python though. Use an actual queue: 
```
from collections import deque
queue = deque(list(start_states))
while queue:
    node = queue.popleft()
```

or, just call it `stack` and `stack.pop()` the last element as the ordering here is irrelevant.

**Review by Karrenbelt — APPROVED** [review]
LGTM

so I don't know how much more checks one might want to add in the future, but its good to be aware that a tool like [NetworkX](https://networkx.org/) or so might come in handy for graph related analyses. It is way too much for what we have here now, just want to point it out as it contains a lot of useful functionalities.

File: docs/autonomy.md [hunk 2]
```diff
@@ -199,7 +199,6 @@ Usage: autonomy analyse abci check-app-specs [OPTIONS]
 Options:
   --check-all          Check all available definitions.
   --packages-dir PATH  Path to packages directory; Use with `--check-all` flag
-  --mermaid            Mermaid file.
```
Comment (DavidMinarsch) [inline docs/autonomy.md:202]: Do we no longer support this?

**Review by DavidMinarsch — APPROVED** [review]

File: docs/autonomy.md (hunk 2 above)
Comment (jmoreira-valory) [reply docs/autonomy.md:202]: We support yaml, json or mermaid for **output**, and yaml json for input. Thus the removal of this line here.

File: autonomy/analyse/abci/app_spec.py (hunk 1 above)
Comment (jmoreira-valory) [reply autonomy/analyse/abci/app_spec.py:227]: Thanks for the advice. Will fix in an upcoming PR.

**Pull request merged**